src/applications/ivc-champva/10-10D/config/submitTransformer.js

```diff
@@ -127,13 +145,23 @@ export default function transformForSubmit(formConfig, form) {
         postalCode: 'NA',
         country: 'NA',
       },
-      dateOfDeath: transformedData?.sponsorDOD || '',
-      dateOfMarriage: transformedData?.sponsorDOM || '',
-      isActiveServiceDeath: transformedData?.sponsorDeathConditions || '',
+      sponsorIsDeceased: transformedData?.sponsorIsDeceased,
+      dateOfDeath: fmtDate(transformedData?.sponsorDOD) || '',
+      // Find the first applicant with a date of marriage to sponsor
+      // (there should only be one) and return that date
+      dateOfMarriage:
+        fmtDate(
+          transformedData?.applicants?.find(
+            a =>
+              a?.applicantRelationshipToSponsor?.relationshipToVeteran ===
+                'spouse' && a?.dateOfMarriageToSponsor !== undefined,
+          )?.dateOfMarriageToSponsor,
+        ) || '',
+      isActiveServiceDeath: transformedData?.sponsorDeathConditions,
     },
     applicants: transformApplicants(transformedData.applicants ?? []),
     certification: {
-      date: transformedData?.dateOfCertification || '',
+      date: fmtDate(new Date().toISOString().replace(/T.*/, '')) || '',
```

Comment:
Previously `dateOfCertification` was just never being set. Replaced it here with the current date in proper format so that we can stamp it next to the certifier signature.